stress/v2/ponyExpress/util.go

```diff
@@ -0,0 +1,88 @@
+package ponyExpress
+
+import (
+	"crypto/rand"
+	"fmt"
+	"strconv"
+	"sync"
+)
+
+// ###########################################
+// ConcurrencyLimiter and associated methods #
+// ###########################################
+
+// ConcurrencyLimiter ensures that no more than a specified
+// max number of goroutines are running.
+type ConcurrencyLimiter struct {
```

Comment:
I stole this code from the old `influx_stress`: https://github.com/influxdata/influxdb/blob/master/stress/util.go#L87